// facebook-plugins/lib/index.js
'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

var _FBComments = require('./FBComments');

var _FBComments2 = _interopRequireDefault(_FBComments);

var _FBCommentsCount = require('./FBCommentsCount');

var _FBCommentsCount2 = _interopRequireDefault(_FBCommentsCount);

var _FBEmbedPost = require('./FBEmbedPost');

var _FBEmbedPost2 = _interopRequireDefault(_FBEmbedPost);

var _FBEmbedVideo = require('./FBEmbedVideo');

var _FBEmbedVideo2 = _interopRequireDefault(_FBEmbedVideo);

var _FBFollow = require('./FBFollow');

var _FBFollow2 = _interopRequireDefault(_FBFollow);

var _FBLike = require('./FBLike');

var _FBLike2 = _interopRequireDefault(_FBLike);

var _FBPage = require('./FBPage');

var _FBPage2 = _interopRequireDefault(_FBPage);

var _FBSend = require('./FBSend');

var _FBSend2 = _interopRequireDefault(_FBSend);

var _FBShare = require('./FBShare');

var _FBShare2 = _interopRequireDefault(_FBShare);

exports['default'] = {
  FBComments: _FBComments2['default'],
  FBCommentsCount: _FBCommentsCount2['default'],
  FBEmbedPost: _FBEmbedPost2['default'],
  FBEmbedVideo: _FBEmbedVideo2['default'],
  FBFollow: _FBFollow2['default'],
  FBLike: _FBLike2['default'],
  FBPage: _FBPage2['default'],
  FBSend: _FBSend2['default'],
  FBShare: _FBShare2['default']
};
module.exports = exports['default'];
